mapper-runtime/src/main/java/com/datastax/oss/driver/api/mapper/annotations/DefaultNullSavingStrategy.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.mapper.annotations;
+
+import com.datastax.oss.driver.api.mapper.entity.saving.NullSavingStrategy;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.CLASS)
+public @interface DefaultNullSavingStrategy {
+  /**
+   * How to handle null entity properties during the insertion.
+   *
+   * <p>If you specify {@code DefaultNullSavingStrategy} at the {@link Dao} level and set one of:
+   * {@link Update#nullSavingStrategy()}, {@link Insert#nullSavingStrategy()}, {@link
+   * SetEntity#nullSavingStrategy()} {@link Query#nullSavingStrategy()} explicitly the method level
+   * annotation value will be taken.
+   *
+   * <p>If you specify {@link DefaultNullSavingStrategy} at the {@link Dao} level and do not set one
+   * of: {@link Update#nullSavingStrategy()}, {@link Insert#nullSavingStrategy()}, {@link
+   * SetEntity#nullSavingStrategy()} {@link Query#nullSavingStrategy()} explicitly the {@code
+   * DefaultNullSavingStrategy#nullSavingStrategy()} will be taken.
+   *
+   * <p>If you do not specify {@code DefaultNullSavingStrategy} at the {@link Dao} level the default
+   * or explicitly user set values from one of: {@link Update#nullSavingStrategy()}, {@link
+   * Insert#nullSavingStrategy()}, {@link SetEntity#nullSavingStrategy()} {@link
+   * Query#nullSavingStrategy()} will be taken
+   */
```

Comment:
I would move this as the main javadoc for the annotation. Also a few suggestions to reorganize the explanations:
```java
/**
 * Annotates a {@link Dao} interface to define a default {@link NullSavingStrategy}, that will apply
 * to all methods that don't explicitly declare one.
 *
 * <p>For example, given this interface:
 *
 * <pre>
 * &#64;Dao
 * &#64;DefaultNullSavingStrategy(SET_TO_NULL)
 * public interface ProductDao {
 *
 *   &#64;Insert
 *   void insert(Product product);
 *
 *   &#64;Update(nullSavingStrategy = DO_NOT_SET)
 *   void update(Product product);
 * }
 * </pre>
 *
 * <ul>
 *   <li>{@code insert(Product)} will use {@link NullSavingStrategy#SET_TO_NULL SET_TO_NULL}
 *       (inherited from the DAO's default).
 *   <li>{@code update(Product)} will use {@link NullSavingStrategy#DO_NOT_SET DO_NOT_SET}.
 * </ul>
 *
 * If the DAO interface isn't annotated with {@link DefaultNullSavingStrategy}, any method that does
 * not declare its own value defaults to {@link NullSavingStrategy#DO_NOT_SET DO_NOT_SET}.
 *
 * <p>Note that null saving strategies are only relevant for {@link Update}, {@link Insert}, {@link
 * Query} and {@link SetEntity} methods.
 */
```